services/npm/npm-downloads.service.js

```diff
@@ -10,95 +10,92 @@ const pointResponseSchema = Joi.object({
 }).required()
 
-// https://github.com/npm/registry/blob/master/docs/download-counts.md#output-1
-const rangeResponseSchema = Joi.object({
-  downloads: Joi.array()
-    .items(pointResponseSchema)
-    .required(),
-}).required()
-
-function DownloadsForInterval(interval) {
-  const { base, messageSuffix = '', query, isRange = false, name } = {
-    week: {
-      base: 'npm/dw',
-      messageSuffix: '/w',
-      query: 'point/last-week',
-      name: 'NpmDownloadsWeek',
-    },
-    month: {
-      base: 'npm/dm',
-      messageSuffix: '/m',
-      query: 'point/last-month',
-      name: 'NpmDownloadsMonth',
-    },
-    year: {
-      base: 'npm/dy',
-      messageSuffix: '/y',
-      query: 'point/last-year',
-      name: 'NpmDownloadsYear',
-    },
-    total: {
-      base: 'npm/dt',
-      query: 'range/1000-01-01:3000-01-01',
-      isRange: true,
-      name: 'NpmDownloadsTotal',
-    },
-  }[interval]
+const intervalMap = {
+  dw: {
+    query: 'point/last-week',
+    schema: pointResponseSchema,
+    transform: json => json.downloads,
+    messageSuffix: '/week',
+  },
+  dm: {
+    query: 'point/last-month',
+    schema: pointResponseSchema,
+    transform: json => json.downloads,
+    messageSuffix: '/month',
+  },
+  dy: {
+    query: 'point/last-year',
+    schema: pointResponseSchema,
+    transform: json => json.downloads,
+    messageSuffix: '/year',
+  },
+  dt: {
+    query: 'range/1000-01-01:3000-01-01',
+    // https://github.com/npm/registry/blob/master/docs/download-counts.md#output-1
+    schema: Joi.object({
+      downloads: Joi.array()
+        .items(pointResponseSchema)
+        .required(),
+    }).required(),
+    transform: json =>
+      json.downloads
+        .map(item => item.downloads)
+        .reduce((accum, current) => accum + current),
+    messageSuffix: '',
+  },
+}
 
-  const schema = isRange ? rangeResponseSchema : pointResponseSchema
+// This hits an entirely different API from the rest of the NPM services, so
+// it does not use NpmBase.
+module.exports = class NpmDownloads extends BaseJsonService {
+  static get category() {
+    return 'downloads'
+  }
 
-  // This hits an entirely different API from the rest of the NPM services, so
-  // it does not use NpmBase.
-  return class NpmDownloads extends BaseJsonService {
-    static get name() {
-      return name
+  static get route() {
+    return {
+      base: 'npm',
+      pattern: ':interval(dw|dm|dy|dt)/:scope(@.+)?/:packageName',
     }
+  }
 
-    static get category() {
-      return 'downloads'
-    }
+  static get examples() {
+    return [
+      {
+        title: 'npm',
+        namedParams: { interval: 'dw', packageName: 'localeval' },
+        staticPreview: this.render({ interval: 'dw', downloadCount: 30000 }),
+        keywords: ['node'],
+      },
+    ]
+  }
 
-    static get route() {
-      return {
-        base,
-        pattern: ':scope(@.+)?/:packageName',
-      }
-    }
+  // For testing.
+  static get _intervalMap() {
```

Comment:
Good idea! This makes things easy to test without having to define (and unnecessarily call) a class level render/transform function